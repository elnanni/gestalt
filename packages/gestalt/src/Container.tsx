import { ReactNode } from 'react';
import Box from './Box';

type Props = {
  /**
   *
   */
  children?: ReactNode;
  /**
   * Available for testing purposes, if needed. Consider [better queries](https://testing-library.com/docs/queries/about/#priority) before using this prop.
   */
  dataTestId?: string;
};

/**
 * [Containers](https://gestalt.pinterest.systems/web/container ) are useful in responsively laying out content on different screens.
 *
 * ![Container light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Container.spec.ts-snapshots/Container-chromium-darwin.png)
 *
 */
export default function Container({ children, dataTestId }: Props) {
  return (
    <Box data-test-id={dataTestId} display="flex" justifyContent="center">
      <Box maxWidth={800} width="100%">
        {children}
      </Box>
    </Box>
  );
}

Container.displayName = 'Container';
