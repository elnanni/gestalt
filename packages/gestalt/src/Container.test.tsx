import renderer from 'react-test-renderer';
import Container from './Container';

test('Container renders', () => {
  const component = renderer.create(<Container />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('validate data test id Container', () => {
  const component = renderer.create(<Container dataTestId='test' />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
