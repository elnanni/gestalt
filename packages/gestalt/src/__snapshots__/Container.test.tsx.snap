// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders 1`] = `
<div
  className="box justifyCenter xsDisplayFlex"
>
  <div
    className="box"
    style={
      Object {
        "maxWidth": 800,
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`validate data test id Container 1`] = `
<div
  className="box justifyCenter xsDisplayFlex"
  data-test-id="test"
>
  <div
    className="box"
    style={
      Object {
        "maxWidth": 800,
        "width": "100%",
      }
    }
  />
</div>
`;
